ganga/GangaLHCb/Lib/Applications/GaudiExec.py

```diff
@@ -309,6 +310,10 @@ def constructExtraFiles(self, job):
             # Add the extra opts file to the job
             tinfo = tarfile.TarInfo(extra_opts_file)
             tinfo.mtime = time.time()
+	    if self.autoDBtags:
+		ddb, conddb = execute('getDDBtags("{0}")'.format(job.inputdata[0].lfn)) # take the tags only from the first file
```

Comment:
I think this function call will need to be cached in some way as it is called for every subjob, but I'll defer to the ganga developers here for advice.

Reply:
@chrisburr The general intention of GaudiExec is that all files are constructed and uploaded before submitting the job so this is done once and stored.
That being said yes it appears this will be called once per-subjob.
Moving this to the configure step and modifying the `self.extraOpts` there should be enough to ensure all subjobs inherit this change. I'd make sure this is correct with a test job but if this breaks let me know but you can always cache the output of this once in the `master_job.application`.

This _will_ crash if a user has forgotten to declare any inputfile(s) here though and it might be nice to catch this in a nicer way.

Reply:
This could fail in several ways so should probably be put into a try except that then throws an exception with a clear explanation. As @rob-c said, a missing dataset, but it could also be an error from Dirac in terms of no information for the given lfn. Also, this will fail if the dataset is not given in terms of `DiracFile` objects.